app.js

```diff
@@ -3,26 +3,68 @@
 const express = require('express');
 const app = express();
 const port = process.env.PORT || 3000;
-const playersData = require('');//should require the data.json file
+const playersData = require('./data.json');//should require the data.json file
 app.use(express.json());
 
 app.get('/players', (req, res) => {
     //should respond with the "players" array inside playersData and Status 200    
+    try {
+        res.status(200).json({
+          data: playersData.players
+        });
+      } catch (err) {
+        res.status(400).json({
+          message: "Error",
+          err
+        });
+      }
 });
 
 app.get('/players/:role', (req, res) => {
     //should respond with only the players that have with the especified role. Status 200.
     //If there's no player with the specified role it should respond with {"error": "No player found"} and Status 404.
+    let { role } = req.params;    
+    try {
+        let playersrole = playersData.players.filter(player => player.role===role);
+        if(playersrole!==[])
+          res.status(200).json({
+            data: playersrole
+          });
+        else
+          res.status(404).json({
+            message: "No player found",
+            err
```

Comment:
the `err` object doesn't exist here. This creates an error, that you catch later, but then we have the 400 error for the 404 case.